connector/dialect.tdd

```diff
@@ -380,60 +151,69 @@
       <argument type='str' />
     </function>
     <function group='cast' name='FLOAT' return-type='real'>
-      <formula>((EXTRACT(EPOCH FROM %1) / ( 60 * 60 * 24)) + (365 * 70 + 17))</formula>
-      <argument type='datetime' />
-    </function>
-    <function group='cast' name='INT' return-type='int'>
-      <formula>(CASE
-	WHEN %1 THEN 1
-	WHEN NOT %1 THEN 0
-	ELSE NULL END)</formula>
+      <formula>CAST(%1 AS DECIMAL)</formula>
       <argument type='bool' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>CAST(TRUNC(%1) AS BIGINT)</formula>
+      <formula>CAST(%1 AS INT)</formula>
       <argument type='real' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>CAST(TRUNC(CAST(%1 AS DOUBLE PRECISION)) AS BIGINT)</formula>
+      <formula>CAST(%1 AS INT)</formula>
+      <argument type='int' />
+    </function>
+    <function group='cast' name='INT' return-type='int'>
+      <formula>CAST(%1 AS INT)</formula>
       <argument type='str' />
     </function>
     <function group='cast' name='INT' return-type='int'>
-      <formula>FLOOR((EXTRACT(EPOCH FROM %1) / ( 60 * 60 * 24)) + (365 * 70 + 17))</formula>
-      <argument type='datetime' />
+      <formula>CAST(%1 AS INT)</formula>
+      <argument type='bool' />
     </function>
     <function group='cast' name='STR' return-type='str'>
       <formula>(CASE
-	WHEN %1 THEN &apos;1&apos;
-	WHEN NOT %1 THEN &apos;0&apos;
-	ELSE NULL END)</formula>
+        WHEN %1 THEN &apos;true&apos;
+        WHEN NOT %1 THEN &apos;false&apos;
+        ELSE NULL END)</formula>
       <argument type='bool' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>CAST(%1 AS TEXT)</formula>
+      <formula>CAST(%1 AS CHAR)</formula>
       <argument type='real' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>CAST(%1 AS TEXT)</formula>
+      <formula>CAST(%1 AS CHAR)</formula>
+      <argument type='int' />
+    </function>
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS CHAR)</formula>
       <argument type='str' />
     </function>
     <function group='cast' name='STR' return-type='str'>
-      <formula>CAST(%1 AS TEXT)</formula>
+      <formula>CAST(%1 AS CHAR)</formula>
+      <argument type='datetime' />
+    </function>
+    <function group='date' name='MONTH' return-type='int'>
+      <formula>CAST(EXTRACT(MONTH FROM %1) AS INTEGER)</formula>
+      <argument type='datetime' />
+    </function>
+    <function group='date' name='NOW' return-type='datetime'>
+      <formula>CURRENT_TIMESTAMP</formula>
+    </function>
+    <function group='date' name='YEAR' return-type='int'>
+      <formula>CAST(EXTRACT(YEAR FROM %1) AS INTEGER)</formula>
       <argument type='datetime' />
     </function>
     <function group='logical' name='IFNULL' return-type='bool'>
```

Comment:
I don't see ifnull in MongoSQL syntax, only nullif. Did I miss it?

Reply:
I'll bring this up at standup, there are some functions that can be implemented in the `formula` such as `IFNULL` and `ZN` even if the actual function itself is not implemented.  The question is whether to do that in order to have more Tableau functions available or wait until it is implemented in mongosql.